packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -209,21 +213,27 @@ export default class ScriptTransformer {
             // files outside `cwd` will not be instrumented
             cwd: this._config.rootDir,
             exclude: [],
+            // Needed for correct coverage as soon as we start storing a source map of the instrumented code
+            inputSourceMap: input.map,
             useInlineSourceMaps: false,
           },
         ],
       ],
+      /**
+       * It's necessary to be able to map back to original source from the instrumented code.
+       * The inline map is needed for debugging functionality, and exposing it as a separate file is needed
+       * for mapping stack traces. It's convenient to use 'both' here and avoid extracting the source map.
+       *
+       * Previous behavior of emitting no map when we can't map back to original source is preserved.
+       */
+      sourceMaps: canMapToInput ? 'both' : false,
     });
 
-    if (result) {
-      const {code} = result;
-
-      if (code) {
-        return code;
-      }
+    if (result && result.code) {
+      return result as TransformResult;
```

Comment:
we _could_ create a type guard function and avoid the cast, but it's not any more type safe. so meh